api/v1beta1/disruption_types.go

```diff
@@ -49,6 +49,7 @@ type DisruptionSpec struct {
 	AdvancedSelector []metav1.LabelSelectorRequirement `json:"advancedSelector,omitempty"` // advanced label selector
 	DryRun           bool                              `json:"dryRun,omitempty"`           // enable dry-run mode
 	OnInit           bool                              `json:"onInit,omitempty"`           // enable disruption on init
+	DurationSeconds  int64                             `json:"durationSeconds"`            // time from disruption creation until chaos pods are deleted and no more are created
```

Comment:
`int64` chosen here so very large seconds values can be used.

Reply:
The field has no `omitempty` json tag here, meaning that it'll be flagged as required, automatically rejecting the disruption if the field is not set. Though we have a mutating webhook setting a default if the field is not set. Should we set this field as optional so it does not force people to set it but we still have the default value from the webhook?

Reply:
Didn't even realize we could take a complex type like that here, I'll switch to Duration.  I am fairly sure that the mutating webhook fires first, so even if people don't set it, it's fine. I want to insist on setting this explicitly as much as possible though?

Reply:
I'll confirm the behavior

Reply:
> I am fairly sure that the mutating webhook fires first

Yes, mutating webhooks are triggered before validating ones

> I want to insist on setting this explicitly as much as possible though?

My main concern is that adding a required field to an existing CRD would break any kind of automation or would break any "saved" disruptions (users keeping a previously working disruption somewhere) while we already have what's needed to apply a default value if not set. Though I understand why you would like it to be explicit, I think adding new fields as optional with a default value is a best practice to avoid breaking changes.